apps/vanilla/src/pages/task2/classes/publishers/turn-generator.ts

```diff
@@ -0,0 +1,28 @@
+import { Publisher } from './publisher';
+
+const PLAYERS_COUNT = 2;
+
+/** Determines whose turn is. */
+export class TurnGenerator extends Publisher<number> {
+
+	private readonly playersCount: number = PLAYERS_COUNT;
+
+	/** Index of a player whose turn is. */
+	public currentPlayerIndex = -1;
+
+	/** Calculates an index of the next player in a queue. */
+	private calcCurrentPlayerIndex = (): void => {
```

Comment:
```suggestion
	private calculateCurrentPlayerIndex = (): void => {
```
See: https://wiki.saritasa.rocks/frontend/languages/ts-js/naming#avoid-contractions